src/Microsoft.VisualStudio.Editor.Razor/DefaultRazorIndentationFactsService.cs

```diff
@@ -63,40 +59,17 @@ internal class DefaultRazorIndentationFactsService : RazorIndentationFactsServic
             }
 
             int? desiredIndentation = null;
-
             SyntaxTreeNode owningChild = owningSpan;
-            while ((owningChild.Parent != null) && !desiredIndentation.HasValue)
+            while (owningChild.Parent != null)
             {
                 var owningParent = owningChild.Parent;
-                var children = new List<SyntaxTreeNode>(owningParent.Children);
-                for (var i = 0; i < children.Count; i++)
+                for (var i = 0; i < owningParent.Children.Count; i++)
                 {
-                    var currentChild = children[i];
-                    if (!currentChild.IsBlock)
+                    var currentChild = owningParent.Children[i];
+                    if (IsCSharpOpenCurlyBrace(currentChild))
                     {
-                        var currentSpan = currentChild as Span;
-                        if (currentSpan.Symbols.Count == 1 &&
-                            currentSpan.Symbols[0] is CSharpSymbol symbol &&
-                            symbol.Type == CSharpSymbolType.LeftBrace)
-                        {
-                            var extraIndent = 0;
-
-                            // Dev11 337312: Only indent one level deeper if the item after the open curly brace is a markup block
-                            if (i < children.Count - 1)
-                            {
-                                var nextChild = children[i + 1];
-                                if (nextChild.IsBlock && ((nextChild as Block).Type == BlockKindInternal.Markup))
-                                {
-                                    extraIndent = indentSize;
-                                }
-                            }
-
-                            // We can't rely on the syntax trees representation of the source document because partial parses may have mutated
-                            // the underlying SyntaxTree text buffer. Because of this, if we want to provide accurate indentations we need to
-                            // operate on the current line representation as indicated by the provider.
-                            var lineText = line.Snapshot.GetLineFromLineNumber(currentSpan.Start.LineIndex).GetText();
-                            desiredIndentation = GetIndentLevelOfLine(lineText, tabSize) + indentSize;
-                        }
+                        var lineText = line.Snapshot.GetLineFromLineNumber(currentChild.Start.LineIndex).GetText();
```

Comment:
It would be great to have some comments documenting the cases we handle and what the expectations are (with examples)